src/jit/codegenxarch.cpp

```diff
@@ -6184,6 +6201,14 @@ void CodeGen::genCallInstruction(GenTreePtr node)
 
 #endif // defined(_TARGET_X86_)
 
+    if (call->IsTailCallViaHelper())
+    {
+        if (compiler->getNeedsGSSecurityCookie())
+        {
+            genEmitGSCookieCheck(true);
+        }
+    }
```

Comment:
The above logic needs to be under #if defined(_TARGET_X86_)?
Note: there is such a #ifdef on line 6191 to 6202, we can include this logic in that.